src/libsystemd/sd-resolve/test-resolve.c

```diff
@@ -30,8 +31,9 @@ static int getaddrinfo_handler(sd_resolve_query *q, int ret, const struct addrin
 
         for (i = ai; i; i = i->ai_next) {
                 _cleanup_free_ char *addr = NULL;
+                sa.sa = *(i->ai_addr);
```

Comment:
no, that's just wrong. the sockaddr can be considerably larger than struct sockaddr

Reply:
Yeah that's the problem with the casts in the first place. I feel like we cornered ourselves here.